doc/SUMMARY.md

```diff
@@ -24,8 +24,11 @@
 - [Tactics](./tactics.md)
 - [Syntax Extensions](./syntax.md)
   - [The `do` Notation](./do.md)
+  - [User-defined notation](./notation.md)
   - [String Interpolation](./stringinterp.md)
 - [Metaprogramming](./metaprogramming.md)
+  - [Macro Overview](./macro_overview.md)
+  - [A Guided Example](./metaprogramming_example.md)
```

Comment:
There is some overlap between the topics, but I think these would better be placed under "Syntax Extensions". We should probably start differentiating between built-in and custom syntax extensions, but that doesn't have to be in this PR. The short `metaprogramming.md` can probably be removed.